python/ray/tests/kuberay/test_autoscaling_config.py

```diff
@@ -35,6 +35,20 @@ def get_basic_ray_cr() -> dict:
     )
     gpu_group["maxReplicas"] = 200
     config["spec"]["workerGroupSpecs"].append(gpu_group)
+    tpu_group = copy.deepcopy(config["spec"]["workerGroupSpecs"][0])
+    tpu_group["groupName"] = "tpu-group"
+    tpu_group["template"]["spec"]["containers"][0]["resources"]["limits"].setdefault(
+        "google.com/tpu", 8
```

Comment:
Why do we set 8 TPU chips for a single Ray worker Pod in the PodSlice here? If I recall correctly, 2x2x2 implies 2 hosts, with each host having 2 TPU cores, and each core containing 2 TPU chips. If we deploy a TPU Pod on a TPU host, it should be 2 * 2 = 4 TPU chips. Do I misunderstand anything?

By the way, I want to confirm another thing: What are the limitations for Pods per TPU host? I remember there are some restrictions, like each TPU host can only have one TPU Pod.

Reply:
This was a typo and is fixed in [08b8eb9](https://github.com/ray-project/ray/pull/43470/commits/08b8eb9016a87d1751461adef7cfd1ae65304494). Each TPU VM host runs on one TPU pod per GKE node. The k8s_resource limits/request for a Pod is expected to consume all of the TPU resources on the GKE node (i.e. we shouldn't have two Pods requesting a subset of TPU chips on the same GKE node). In the test example, there are two 2 hosts each with 4 TPU chips, so the `google.com/tpu` value should be 4.